backend/src/v4/models/modelSetting.js

```diff
@@ -438,33 +445,42 @@ ModelSetting.updateMultiplePermissions = async function(account, modelIds, updat
 };
 
 ModelSetting.updatePermissions = async function(account, model, permissions = []) {
-	const setting = await ModelSetting.findModelSettingById(account, model);
+	const { findByUserName, teamspaceMemberCheck } = require("./user");
 
+	if (!Array.isArray(permissions)) {
+		throw responseCodes.INVALID_ARGUMENTS;
+	}
+
+	const setting = await ModelSetting.findModelSettingById(account, model);
 	if (!setting) {
 		throw responseCodes.MODEL_NOT_FOUND;
 	}
 
-	permissions.forEach((permissionUpdate) => {
-		if (!setting.permissions) {
-			setting.permissions = [];
-		}
-
-		const userIndex = setting.permissions.findIndex(x => x.user === permissionUpdate.user);
+	permissions = _.uniq(permissions, "user");
 
-		if (-1 !== userIndex) {
-			if ("" !== permissionUpdate.permission) {
-				setting.permissions[userIndex].permission = permissionUpdate.permission;
-			} else {
-				setting.permissions.splice(userIndex, 1);
+	const promises = [];
+	const dbUser = await findByUserName(account);
+	permissions.forEach(permission => {
+		checkPermissionIsValid(permission);
+		checkUserHasPermissionTemplate(dbUser, permission);
+
+		promises.push(teamspaceMemberCheck(permission.user, dbUser.user).then(() => {
```

Comment:
use async/await syntax instead of .then 

```suggestion
		promises.push(teamspaceMemberCheck(permission.user, dbUser.user).then(() => {
```